File: backend/internal/dto/response/withdrawal.go
package response

import "time"

type WithdrawalOrderResp struct {
	ID                 uint64    `json:"id"`
	Type               string    `json:"type"`
	Currency           string    `json:"currency"`
	Chain              *string   `json:"chain,omitempty"`
	Amount             string    `json:"amount"`
	PlatformFee        string    `json:"platform_fee"`
	FeeDeductionMethod string    `json:"fee_deduction_method"`
	NetAmount          string    `json:"net_amount"`
	Status             string    `json:"status"`
	ReviewStatus       string    `json:"review_status"`
	ToAddress          *string   `json:"to_address,omitempty"`
	TxID               *string   `json:"tx_id,omitempty"`
	CreatedAt          time.Time `json:"created_at"`
}

type WithdrawalOrderListResp struct {
	Orders []WithdrawalOrderResp `json:"orders"`
	Total  int64                 `json:"total"`
}

type WithdrawalFeePreviewResp struct {
	Currency           string `json:"currency"`
	Amount             string `json:"amount"`
	PlatformFee        string `json:"platform_fee"`
	FeeDeductionMethod string `json:"fee_deduction_method"`
	TotalDeduction     string `json:"total_deduction"`
	NetAmount          string `json:"net_amount"`
}

File: admin/src/types/withdrawal.ts
export interface AdminWithdrawal {
  id: number;
  platform_order_id: string;
  merchant_id: number;
  merchant_email: string;
  type: string;
  currency: string;
  network?: string;
  amount: string;
  platform_fee: string;
  status: string;
  review_status: string;
  to_address?: string;
  tx_id?: string;
  created_at: string;
  completed_at?: string;
}

export interface AdminWithdrawalListResponse {
  withdrawals: AdminWithdrawal[];
  total: number;
}
